pkg/providers/tinkerbell/validator_helper_test.go

```diff
@@ -0,0 +1,8 @@
+package tinkerbell
+
+import "github.com/aws/eks-anywhere/pkg/hardware"
+
+// SetValidatorHardwareConfig sets v internal hardwareConfig member to cfg.
+func SetValidatorHardwareConfig(v *Validator, cfg hardware.HardwareConfig) {
+	v.hardwareConfig = cfg
+}
```

Comment:
i would vote to just do this inline instead of having a separate function for this

Reply:
If you look at the packages they're different - I've used the `_test` package idiom for the unit tests. This is a helper function used in testing only, because it shouldn't be a part of the package API, for hacking a `Validator` instance removing the need to construct the `Validator` in full. It exists largely because of a tight coupling between reading hardware configuration and one of the validate methods that lazily loads the hardware config. As a rule of thumb you don't want io occurring in unit tests if you can help it. 

In short, this can't be done in-line using the `_test` idiom which is best practice for unit testing with some small exceptions.